library/common.h

```diff
@@ -57,4 +59,255 @@
  */
 #define MBEDTLS_ALLOW_PRIVATE_ACCESS
 
+/** Byte Reading Macros
+ *
+ * Obtain the most significant byte of x using 0xff
```

Comment:
Those comments are not really correct: why the most significant byte ? why to retrieve the "next" byte ? Please improve them.

Reply:
Yes you are right, How does this look?
```
 * Given a data \p x , retrieve a specified byte from x.
 * Each macro shifts the data by x bytes such that the xth byte can 
 * masked out, as the least significant byte, using & 0xff
```

Reply:
Rather than "data" I would use "multi-byte integer". Then the second part goes in to much details I would say.
Thus I would suggest something like:
"Given a multi-byte integer \p x, MBEDTLS_BYTE_n retrieves the n-th byte from x, where byte 0 is the least significant byte."